### ISSUE TITLE
[Bug]: package does not use set locale after it's cached the previous value

### ISSUE THREAD

**Issue by PavelSavushkinMix** [Laravel-Lang/common#180]
### Lang Package Name

laravel-lang/locales

### Lang Package Version

locales: 2.6.0; lang: 15.1.4

### Laravel Framework Version

11.0.7

### PHP Version

8.3.3

### Issue description

If the package/application calls `\LaravelLang\Locales\Facades\Locales::getDefault`, setting a new locale for the application does not affect the further `Locales` calls.

Code to test:
```php
Route::get('test', function () {
    dump(app()->getLocale()); // Let's say, "en" by default
    // \LaravelLang\Locales\Facades\Locales::getDefault();
    app()->setLocale('de');
    dd(\LaravelLang\Locales\Facades\Locales::info('de'));
});
```
The code above returns the correct localized name for the locale (`Deutsch`). If we decomment the 3d line (`getDefault`), the latest dump would provide `German`, even if the locale is set to `de` on the line above. My guess, that relates to the next implementation:
`vendor/laravel-lang/locales/src/Services/RawLocales.php :: 124 line`

```php
public function getDefault(): string
{
    return $this->registry(__METHOD__, function () {
        $locale = config('app.locale');

        return $this->toAlias(
            $this->isAvailable($locale) ? $locale : $this->getFallback()
        );
    });
}
```
While we are using the `__METHOD__` to cache the results, it binds the first value of the application's locale (`app()->getLocale()`). I thought I should use the package's `vendor/laravel-lang/locales/src/Services/Locales::set`, but that does not solve the issue, because there is no any cache invalidation.

### Steps to reproduce

```php
Route::get('test', function () {
    dump(app()->getLocale()); // Let's say, "en" by default
    // \LaravelLang\Locales\Facades\Locales::getDefault();
    app()->setLocale('de');
    dd(\LaravelLang\Locales\Facades\Locales::info('de'));
});
```

**Steps to reproduce:**
1. Run the code.
2. See the correct `localized` field for the `Locales::info`' result.
3. Decomment `Locales::getDefault` line.
4. Run the code.

**Actual:**
The `localized` field contains the original default locale of the application.

**Expected:**
The `localized` field contains the correct translated language name according to the set locale before.

**Comment by andrey-helldar** [comment]
Released in version [2.7.0](https://github.com/Laravel-Lang/locales/releases/tag/2.7.0).

Thanks for your issue!

**Comment by PavelSavushkinMix** [comment]
> Released in version [2.7.0](https://github.com/Laravel-Lang/locales/releases/tag/2.7.0).
> 
> Thanks for your issue!

Thanks for fixing it quickly! :)
